extensions/cba_settings/cba_settings.cpp

```diff
@@ -0,0 +1,335 @@
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <algorithm>
+#include "trim.h"
+
+#define VERSION "1.0.0"
+
+#define RET \
+strncpy_s(output, outputSize, result.c_str(), _TRUNCATE);\
+return
+
+#define RETURN(val)\
+strncpy_s(output, outputSize, result.c_str(), _TRUNCATE);\
+return (val)
+
+#define SYNTAX_ERROR_WRONG_PARAMS_SIZE 101
+#define SYNTAX_ERROR_WRONG_PARAMS_TYPE 102
+#define PARAMS_ERROR_TOO_MANY_ARGS 201
+#define EXECUTION_WARNING_TAKES_TOO_LONG 301
+
+#define TEMP_FILE "userconfig/temp"
+#define SETTINGS_FILE "userconfig/cba_settings.sqf"
+#define DELIM ":"
+#define MAX_CALLBACK_LENGTH 10000
+
+using namespace std;
+
+string version() {
+    return VERSION;
+}
+
+string clear() {
+    ofstream file;
+    file.open(SETTINGS_FILE, ios::out);
+    file.close();
+    return "true";
+}
+
+string load(int strlen) {
+    string result = "";
+
+    ifstream file;
+    file.open(SETTINGS_FILE, ios::in);
+
+    string line;
+    while (file.good()) {
+        getline(file, line);
+        if (file.good() || line.length() != 0) // Don't explode file by appending duplicate additional newlines.
+            result = result + trim(line) + "\n";
+    }
+
+    file.close();
+
+    if (strlen * MAX_CALLBACK_LENGTH > result.length())
+        return "";
+
+    result = result.substr(strlen * MAX_CALLBACK_LENGTH, MAX_CALLBACK_LENGTH);
+    return result;
+}
+
+string parse(int strlen) {
+    string result = "[";
+
+    ifstream file;
+    file.open(SETTINGS_FILE, ios::in);
+
+    string line;
+    while (file.good()) {
+        getline(file, line);
+        line = trim_left(line);
+
+        // Ignore commented out lines.
+        if (line.substr(0, 2) == "//")
```

Comment:
This copies the string every time. Would be WAY more efficient to use a string_view here, as that is all that you need. For trim_left too.

Reply:
Also why trim left and right seperately? why not immediately trim both sides?

Reply:
>why not immediately trim both sides?

```
    force force   setting    =    123;
```
is something one could write. If I process this, I need multiple trims, but not on both sides all the time.